pom.xml

```diff
@@ -209,6 +209,19 @@
         </configuration>
       </plugin>
 
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <version>3.0.0-M5</version>
+        <configuration>
+          <excludes>
+            <exclude>**/*IntegrationTests</exclude>
```

Comment:
Junit5 has some interesting annotations that can do this as part of the test suite. What do you think about updating to it? 

Something like [DisabledIfEnvironmentVariable](https://junit.org/junit5/docs/5.2.0/api/org/junit/jupiter/api/condition/DisabledIfEnvironmentVariable.html)?

Reply:
Overall I think updating to JUnit5 is a good idea.  I'm not sure if it would break anything in the existing unit tests for CoreSocketFactory, so it might be out of the scope of this PR.